### ISSUE TITLE
Heat map + Offline exporting: The legend is black when exporting PDF

### ISSUE THREAD

**Issue by hattn125** [highcharts/highcharts#20332]
#### Actual behaviour
In heatmap, when using offline export by PDF: The legend is always black

![image](https://github.com/highcharts/highcharts/assets/145006963/e7b5c8e3-99ef-47b9-8fc6-47a82a6467a8)

#### Expected behaviour
In heatmap, the offline exporting by PDF should produce the legend with its color properly.

#### Live demo with steps to reproduce
https://jsfiddle.net/xhvzn2ag/85/

#### Product version
11.2.0

#### Affected browser(s)
All

**Comment by karolkolodziej** [comment]
Hi @hattn125!
Thank you for reporting the issue! 

There are warnings in the console about missing tags/attributes in the AST. If you add them everything works as expected.
[See the demo.
](https://jsfiddle.net/BlackLabel/ejsd23uL/)
```js
Highcharts.AST.allowedAttributes.push('filter','stop-color','stop-opacity');
Highcharts.AST.allowedTags.push('lineargradient','fedropshadow');
```

@TorsteinHonsi should we treat it as a bug and add the missing parts to the AST?

**Comment by TorsteinHonsi** [comment]
@karolkolodziej Yes I definately think we should treat this as a bug, since the configuration is just plain Highcharts without any hacks, and it fails.

For the `allowedAttributes`, I think we should just add those to the list.

The `allowedTags` however are different. We already have `linearGradient` and `feDropShadow` in the list. SVG is case sensitive, and if we export the chart to SVG and change `linearGradient` to `lineargradient`, it doesn't work. So this smells like a bug somewhere in either in the offline exporting module or svg2pdf/jspdf. Why is it changed to lower case? Can we fix that? Or change it back to camel case before being fed into the AST?

**Comment by hattn125** [comment]
Thanks guys for answering! 
I'm looking forward an offical fix :)
